### ISSUE TITLE
Problems with reproducing the results in this paper

### ISSUE THREAD

**Issue by SherlockEddie** [Perturbation-Response-Prediction/PRnet#9]
Hello, I have learned a lot of useful knowledge by reading your article. In order to deepen my understanding of this article, I want to try to use these open source codes you provided to reproduce the relevant results of this paper, but the results I finally got seem to be quite different from the paper, so I would like to ask how to use these data and codes to reproduce. Let me describe how I reproduced it: First, I downloaded the latest source data and source code you provided. In the checkpoint folder, I found the model pretrained weight(sciplex_best_epoch_all.pt), so I directly ran test_sciplex.py and analysis_sciplex.py to get the results. I don't know whether the model needs to be retrained or whether my steps to reproduce are wrong. I am very eager to know how to reproduce the results of this paper. I hope to get your help. Thank you very much!

**Comment by nicole1q** [comment]
#### Pretrained Weight
The file sciplex_best_epoch_all.pt represents the result of one fold in a five-fold cross-validation. To reproduce the results in the paper more comprehensively, you would need to re-train the model using the full dataset and perform five-fold cross-validation. This can be done by modifying the split_key parameter during training to ensure proper data splitting for each fold.
#### Reproducing Results
To ensure alignment with the paper’s results:
1.Retrain the model using train_sciplex.py with your specific dataset and ensure the split_key is appropriately adjusted for each fold.
2.Average the results from all five folds to match the overall results reported in the paper.
3.Make sure that any configuration files or hyperparameters align with those specified in the repository or the supplementary materials of the paper.
#### Differences in Results
If your reproduced results differ from the paper, could you please describe in detail the specific discrepancies you observed? For example, are there differences in performance metrics (e.g., accuracy, correlation), output files, or patterns in predictions?

**Comment by Flroa10086633** [comment]
Hello. I also encountered the same problem. I downloaded the latest data and code, and directly ran:

`python test_lincs.py --split_key drug_split_4 `
`python analysis_lincs.py --split_key drug_split_4 `

It seems to use the weights provided here: checkpoint/lincs_best_epoch_all.pt

However, I obtained the following results:

> ,split_key,condition_r2_score,foldchange_pearson
> 0,drug_split_4,-1.4878103444503996,-0.04376220798455383
> 1,drug_split_4,-1.4878103444503996,0.07703289677689379

This weight seems to be incorrect. Additionally, could you provide model weights for other folds? 
Thanks

**Comment by Flroa10086633** [comment]
Additionally, I tried to retrain PRNet on Lincs (one 4090 GPU, 500 epochs, 512 batch size), but it took over 20 days! @nicole1q

**Comment by ZexiChen502** [comment]
> Additionally, I tried to retrain PRNet on Lincs (one 4090 GPU, 500 epochs, 512 batch size), but it took over 20 days! @nicole1q

You don't have to try, I've replicated the model and found this article to be of limited value. I optimized Dataloader so that it now takes only a few minutes to complete an epoch(it used to take hours), and retrained the model with more reasonable evaluation metrics (it didn't), as the metrics didn't match the published version.

**Comment by nicole1q** [comment]
> Hello. I also encountered the same problem. I downloaded the latest data and code, and directly ran:
> 
> `python test_lincs.py --split_key drug_split_4 ` `python analysis_lincs.py --split_key drug_split_4 `
> 
> It seems to use the weights provided here: checkpoint/lincs_best_epoch_all.pt
> 
> However, I obtained the following results:
> 
> > ,split_key,condition_r2_score,foldchange_pearson
> > 0,drug_split_4,-1.4878103444503996,-0.04376220798455383
> > 1,drug_split_4,-1.4878103444503996,0.07703289677689379
> 
> This weight seems to be incorrect. Additionally, could you provide model weights for other folds? Thanks

Thank you for bringing this to our attention and for your detailed description of the issue.

To reproduce results consistent with the paper, please follow these steps:

1. **Retrain the Model**:
   Use `train_lincs.py` or `train_sciplex.py` with your specific dataset. Ensure that the `split_key` is appropriately set for the desired fold. Retraining will ensure compatibility with your setup and data.
2. **Verify Configurations**:
   Check that all configuration files and hyperparameters match those provided in the repository or the supplementary materials of the paper. Misalignments in settings can lead to discrepancies in results.
3. **Adjust `obs_key`**:
   Change the `obs_key` to `cov_drug_name`  or  `cov_drug` in the `test_lincs.py`  or `test_sciplex.py` to ensure the correct mapping of conditions.
4. **Preprocessing**:
   Make sure to apply normalization and log-transformation to your data before running `test_lincs.py` or `test_sciplex.py`. This step is crucial for achieving accurate predictions.
5. **Compounds vs. Cov-Drug Analysis**:
   The script `analysis_lincs.py` calculates results based on `cov-drug` condition. If you wish to analyze compound-level results, you may need to adjust the calculation method accordingly.

We have recently updated the GitHub repository. Please pull the latest version, which should resolve these issues.

**Comment by ZexiChen502** [comment]
> > Hello. I also encountered the same problem. I downloaded the latest data and code, and directly ran:
> > `python test_lincs.py --split_key drug_split_4 ` `python analysis_lincs.py --split_key drug_split_4 `
> > It seems to use the weights provided here: checkpoint/lincs_best_epoch_all.pt
> > However, I obtained the following results:
> > > ,split_key,condition_r2_score,foldchange_pearson
> > > 0,drug_split_4,-1.4878103444503996,-0.04376220798455383
> > > 1,drug_split_4,-1.4878103444503996,0.07703289677689379
> > 
> > 
> > This weight seems to be incorrect. Additionally, could you provide model weights for other folds? Thanks
> 
> Thank you for bringing this to our attention and for your detailed description of the issue.
> 
> To reproduce results consistent with the paper, please follow these steps:
> 
> 1. **Retrain the Model**:
>    Use `train_lincs.py` or `train_sciplex.py` with your specific dataset. Ensure that the `split_key` is appropriately set for the desired fold. Retraining will ensure compatibility with your setup and data.
> 2. **Verify Configurations**:
>    Check that all configuration files and hyperparameters match those provided in the repository or the supplementary materials of the paper. Misalignments in settings can lead to discrepancies in results.
> 3. **Adjust `obs_key`**:
>    Change the `obs_key` to `cov_drug_name`  or  `cov_drug` in the `test_lincs.py`  or `test_sciplex.py` to ensure the correct mapping of conditions.
> 4. **Preprocessing**:
>    Make sure to apply normalization and log-transformation to your data before running `test_lincs.py` or `test_sciplex.py`. This step is crucial for achieving accurate predictions.
> 5. **Compounds vs. Cov-Drug Analysis**:
>    The script `analysis_lincs.py` calculates results based on `cov-drug` condition. If you wish to analyze compound-level results, you may need to adjust the calculation method accordingly.
> 
> We have recently updated the GitHub repository. Please pull the latest version, which should resolve these issues.

Why do you calculate the R² by averaging the predicted values and true values for all samples in a batch, rather than calculating R² for the predicted values and true values of the entire sample?  It can be problematic.

**Comment by Biolg** [comment]
> > Additionally, I tried to retrain PRNet on Lincs (one 4090 GPU, 500 epochs, 512 batch size), but it took over 20 days! [@nicole1q](https://github.com/nicole1q)
> 
> You don't have to try, I've replicated the model and found this article to be of limited value. I optimized Dataloader so that it now takes only a few minutes to complete an epoch(it used to take hours), and retrained the model with more reasonable evaluation metrics (it didn't), as the metrics didn't match the published version.

Hi, I also tried reproducing PRNet and encountered the same training inefficiency. Your optimization sounds very helpful — would you mind sharing the improved DataLoader or training scripts for reference? Thanks a lot!

**Comment by lian-xiao** [comment]
> > > Hello. I also encountered the same problem. I downloaded the latest data and code, and directly ran:
> > > `python test_lincs.py --split_key drug_split_4 ` `python analysis_lincs.py --split_key drug_split_4 `
> > > It seems to use the weights provided here: checkpoint/lincs_best_epoch_all.pt
> > > However, I obtained the following results:
> > > > ,split_key,condition_r2_score,foldchange_pearson
> > > > 0,drug_split_4,-1.4878103444503996,-0.04376220798455383
> > > > 1,drug_split_4,-1.4878103444503996,0.07703289677689379
> > > 
> > > 
> > > This weight seems to be incorrect. Additionally, could you provide model weights for other folds? Thanks
> > 
> > 
> > Thank you for bringing this to our attention and for your detailed description of the issue.
> > To reproduce results consistent with the paper, please follow these steps:
> > 
> > 1. **Retrain the Model**:
> >    Use `train_lincs.py` or `train_sciplex.py` with your specific dataset. Ensure that the `split_key` is appropriately set for the desired fold. Retraining will ensure compatibility with your setup and data.
> > 2. **Verify Configurations**:
> >    Check that all configuration files and hyperparameters match those provided in the repository or the supplementary materials of the paper. Misalignments in settings can lead to discrepancies in results.
> > 3. **Adjust `obs_key`**:
> >    Change the `obs_key` to `cov_drug_name`  or  `cov_drug` in the `test_lincs.py`  or `test_sciplex.py` to ensure the correct mapping of conditions.
> > 4. **Preprocessing**:
> >    Make sure to apply normalization and log-transformation to your data before running `test_lincs.py` or `test_sciplex.py`. This step is crucial for achieving accurate predictions.
> > 5. **Compounds vs. Cov-Drug Analysis**:
> >    The script `analysis_lincs.py` calculates results based on `cov-drug` condition. If you wish to analyze compound-level results, you may need to adjust the calculation method accordingly.
> > 
> > We have recently updated the GitHub repository. Please pull the latest version, which should resolve these issues.
> 
> Why do you calculate the R² by averaging the predicted values and true values for all samples in a batch, rather than calculating R² for the predicted values and true values of the entire sample? It can be problematic.

Hello, I also encountered difficulties when reproducing PRnet. May I ask if you successfully reproduced it？
